2-mandatory/2-financial-times.js

```diff
@@ -4,36 +4,90 @@
     The home page of the web site has a headline section, which only has space for article titles which are 65 characters or less.
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
-function potentialHeadlines(allArticleTitles) {
-    // TODO
-}
 
+    function potentialHeadlines(allArticleTitles) {
+        // Create an empty array to store the potential headlines
+        const potentialHeadlines = [];
+      
+        // Loop through each article title in the input array
+        for (let i = 0; i < allArticleTitles.length; i++) {
+          const articleTitle = allArticleTitles[i];
+      
+          // Check if the title is 65 characters or less
+          if (articleTitle.length <= 65) {
+            // If it is, add it to the potential headlines array
+            potentialHeadlines.push(articleTitle);
+          }
+        }
+      
+        // Return the array of potential headlines
+        return potentialHeadlines;
+      }
+      
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
-    (you can assume words will always be seperated by a space)
+    (you can assume words will always be separated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
-}
+   
+    let fewestWords = allArticleTitles[0];
+  
+    for (let i = 0; i < allArticleTitles.length; i++) {
+      if (fewestWords.length > allArticleTitles[i].length) {
+            fewestWords = allArticleTitles[i];
+      }
+    }
+  
+    return fewestWords;
+  }
+  
 
 /*
-    The editor of the FT has realised that headlines which have numbers in them get more clicks!
+    The editor of the FT has realized that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
+
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
-}
+    const headlinesWithNumbers = [];
+  
+    for (let i = 0; i < allArticleTitles.length; i++) {
+      const articleTitle = allArticleTitles[i];
+      
+      for (let j = 0; j < articleTitle.length; j++) {
+        const character = articleTitle.charAt(j);
+        
+        if (!isNaN(parseInt(character))) {
+          headlinesWithNumbers.push(articleTitle);
+        }
+      }
+    }
+  
+    return headlinesWithNumbers;
+  }
```

Comment:
Comment: Interesting. I see you used .charAt and NaN. I have not seen any other use it. 

My solution:
  let newArr = [];
  for (let title of allArticleTitles) {
    if (/[0-9]/.test(title) === true) {
      newArr.push(title);
    }
  }

  return newArr;
}